library/src/google-maps-provider.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {useState, useEffect, PropsWithChildren} from 'react';
+
+// https://developers.google.com/maps/documentation/javascript/url-params
+export interface GoogleMapsAPIUrlParameters {
+  libraries?: string[];
+  language?: string;
+  region?: string;
+  version?: string;
+  authReferrerPolicy?: string;
+}
+
+export interface GoogleMapsProviderProps extends GoogleMapsAPIUrlParameters {
+  googleMapsAPIKey: string;
+  mapContainer?: HTMLElement | null;
```

Comment:
We could all of the others that are no `mapOptions` and no `URLParameters` in one type called `mapConfig` (maybe even the mapOptions). But just an idea, works also like this.

Reply:
I think I meant "name" here...